app/src/main/java/ca/klapstein/baudit/presenters/CreateAccountPresenter.java

```diff
@@ -18,59 +24,50 @@ public CreateAccountPresenter(CreateAccountView view, Context context) {
     }
 
     public void confirmButtonClicked(int checkedId, String firstName, String lastName,
-                                     String username, String email, String phoneNumber) {
+                                     String usernameString, String email, String phoneNumber) {
         boolean validAccount = true;
         view.updateUsernameError("");
         view.updateEmailError("");
         view.updatePhoneNumberError("");
 
-        Username newUsername = null;
-        try {
-            newUsername = new Username(username);
-            if (!dataManager.uniqueID(newUsername)) {
-                Log.e(TAG, "given non unique userID");
-                validAccount = false;
-                view.updateUsernameError(context.getResources().getString(R.string.username_error));
-            }
-        } catch (IllegalArgumentException e) {
-            Log.e(TAG, "given invalid userID: " + username, e);
+        if (!Username.isValid(usernameString)) {
             validAccount = false;
             view.updateUsernameError(context.getResources().getString(R.string.username_error));
         }
 
-        Email newEmail = null;
-        try {
-            newEmail = new Email(email);
-        } catch (IllegalArgumentException e) {
+        if (!new DataModel(this.context).uniqueID( new Username(usernameString))) {
```

Comment:
can't we use the local instance of `dataManager` as itself is a `DataModel` class?